gr-fec/lib/async_decoder_impl.cc

```diff
@@ -112,88 +107,56 @@ void async_decoder_impl::decode_unpacked(pmt::pmt_t msg)
 
     size_t o0(0);
     const float* f32in = pmt::f32vector_elements(bits, o0);
-    pmt::pmt_t outvec(pmt::make_u8vector(nbits_out, 0x00));
+    const size_t nitems_out = d_packed ? nbits_out / 8 : nbits_out;
+    pmt::pmt_t outvec(pmt::make_u8vector(nitems_out, 0x00));
     uint8_t* u8out = pmt::u8vector_writable_elements(outvec, o0);
+    uint8_t* decoder_out = d_packed ? d_bits_out.data() : u8out;
+    const float shift = d_decoder->get_shift();
 
-    if (strncmp(d_decoder->get_input_conversion(), "uchar", 5) == 0) {
+    if (std::string_view(d_decoder->get_input_conversion()) == "uchar") {
+        // TODO: replace all this by volk_32f_s32f_x2_convert_8u when it gets
+        // into a stable volk release
         volk_32f_s32f_multiply_32f(d_tmp_f32.data(), f32in, 48.0f, nbits_in);
-    } else {
-        memcpy(d_tmp_f32.data(), f32in, nbits_in * sizeof(float));
-    }
-
-    if (d_decoder->get_shift() != 0) {
-        for (size_t n = 0; n < nbits_in; n++)
-            d_tmp_f32[n] += d_decoder->get_shift();
-    }
-
-    if (strncmp(d_decoder->get_input_conversion(), "uchar", 5) == 0) {
-        // volk_32f_s32f_convert_8i(d_tmp_u8, d_tmp_f32, 1, nbits_in);
-        for (size_t n = 0; n < nbits_in; n++)
-            d_tmp_u8[n] = static_cast<uint8_t>(d_tmp_f32[n]);
+        if (shift != 0.0) {
```

Comment:
I was wondering why a `shift` variable is of type `float`. Looing at L114, I assume the name `bias` would've been more fitting.
I'm torn between updating the variable name and leaving it as is because of consistency.

Reply:
Yes. In hindsight `bias` seems a better name than `shift`. However, this value comes from `d_decoder->get_shift()`, so unless we want to rename that to `get_bias()` in all the decoders, I think it's better to call the variable `shift`.

Reply:
On the other hand, due to applying https://github.com/gnuradio/gnuradio/pull/6537#discussion_r1115398262, this variable can disappear (we can put the call to `d_decoder->get_shift()` directly as a function parameter), so matter solved.